xayn-ai/src/ltr/list_net/mod.rs

```diff
@@ -664,15 +652,18 @@ where
             .next_evaluation_sample()
             .map_err(TrainingError::Data)?
         {
-            let (scores_y, _) = list_net.calculate_intermediate_scores(inputs, false);
-            let prob_dist_y = list_net.calculate_final_scores(&scores_y);
-            costs.push(cost_function(target_prob_dist, prob_dist_y.view()));
+            let (scores_y, _) = self.list_net.calculate_intermediate_scores(inputs, false);
+            let prob_dist_y = self.list_net.calculate_final_scores(&scores_y);
+            let cost = cost_function(target_prob_dist, prob_dist_y.view());
+            self.callbacks
+                .evaluation_result(cost)
+                .map_err(TrainingError::Control)?;
         }
 
-        let count = costs.len() as f32;
-        let mean =
-            (count > 0.).then(|| costs.into_iter().fold(0f32, |acc, cost| acc + cost / count));
-        Ok(mean)
+        self.callbacks
+            .end_of_evaluation()
+            .map_err(TrainingError::Control)?;
+        Ok(())
```

Comment:
```suggestion
            .map_err(TrainingError::Control)
```

Reply:
Can I keep it the way it is?

We semantically return no error, and changing it to return the `Ok` value of `end_of_evaluation()` does to me convey "we return whatever `.end_of_evaluation()`" returns (which just happens to be `()`).

Reply:
I'm fine keeping it that way 👍